gcp/gcp.go

```diff
@@ -89,6 +94,39 @@ func (b *Backend) ReadFile(ctx context.Context, readBucket string, readLocation
 	return data, nil
 }
 
+func (b *Backend) GetNodeId(ctx context.Context) string {
+	return os.Getenv("GAE_INSTANCE")
+}
+
+func (b *Backend) GetDeploymentId(ctx context.Context) string {
+	return os.Getenv("GAE_DEPLOYMENT_ID")
+}
+
+func (b *Backend) WriteLeaderFile(ctx context.Context, metadataBucket string, nodeId string, deploymentId string) error {
```

Comment:
Since it's writing a struct that's in `firehose` package, should this function just accept that struct instead of all this info

Reply:
oh an then the firehose pkg puts the data in that struct instead of this function?